src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/advice/ApplicationProblemControllerAdvice.java

```diff
@@ -58,10 +60,16 @@
 })
 @RequestMapping(produces = APPLICATION_PROBLEM_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
 @SuppressWarnings({"PMD.ExcessiveImports", "PMD.DataflowAnomalyAnalysis",
-    "PMD.UseStringBufferForStringAppends", "PMD.LawOfDemeter"})
+    "PMD.UseStringBufferForStringAppends", "PMD.LawOfDemeter", "PMD.CouplingBetweenObjects"})
 public class ApplicationProblemControllerAdvice extends BaseControllerAdvice implements ValidationAdviceTrait {
 
-    @ExceptionHandler(FeignException.ServiceUnavailable.class)
+    @ExceptionHandler({
+        FeignException.ServiceUnavailable.class,
+        FeignException.FeignServerException.class,
+        FeignException.GatewayTimeout.class,
+        FeignException.BadRequest.class,
+        ServerErrorException.class,
+    })
     public ResponseEntity<ThrowableProblem> handleFeignServiceUnavailableException(FeignException ex) {
```

Comment:
Should probably rename this method now as it deals with more exceptions than before.

Reply:
renamed